python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_condition.py

```diff
@@ -263,9 +265,22 @@ def create(
         context: "SchedulingContext",
         true_slice: AssetSlice,
         subsets_with_metadata: Sequence[AssetSubsetWithMetadata] = [],
+        slices_with_metadata: Sequence[Tuple[AssetSlice, MetadataMapping]] = [],
```

Comment:
I think we should introduce a new object here that capture an AssetSlice enrich with metadata, rather than relying on a tuple.